### ISSUE TITLE
[Q] How to set output volume?

### ISSUE THREAD

**Issue by NightMachinary** [overtone/overtone#472]
E.g., in this example:
```
(demo 1
      (let [vals (dwhite 0 15 INF)
            trig (impulse:kr (mouse-x 1 40 1))
            val (demand:kr trig 0 vals)
            poll (poll trig val "diwhite val:")
            freq (+ 340 (* 30 val))]
        (* 0.1 (sin-osc freq))))
```

**Comment by harold** [comment]
If I understand you correctly, the `0.1` there is the volume (amplitude attenuation on the oscillator output). Bigger numbers will make it louder, smaller numbers will make it quieter.

**Comment by NightMachinary** [comment]
@harold You're right. I actually guessed this, but I only tried increasing it to 10, which didn't have that much effect. Increasing it to `30000` shows somewhat of an effect. 

Can't this be done generally?

**Comment by harold** [comment]
In general it will depend a lot on the incoming amplitude (before multiplication)

30000 IS A LOT - be careful (!), these values go straight to your speakers.

`(* 0.5 ...)` would be half the amplitude and `(* 2 ...)` would be double the amplitude.

If you needed to crank it that much to hear it, it may be that what you're multiplying is a lot smaller than you think it is.

Perhaps try with some simpler examples (without dwhite, impulse, demand, mouse-*, etc...) to get a less convoluted idea of what's going on.

**Comment by NightMachinary** [comment]
I was thinking of something simple like how media players can change the volume 😅 (which, admittedly, is much better when lowering the volume than making it louder).

**Comment by harold** [comment]
Yeah, that's not precisely the level of abstraction that overtone (SC) operates at. 🤭

**Comment by plexus** [comment]
There is actually a `volume` function, which can either get or set the global volume for Overtone

```clj
(volume) ;;=> 0.8
(volume 0.7)
```

If you use `definst` (instead of `defsynth`) you can use `inst-volume!` to control the volume of the inst, also while it's playing. So you could even hook that up to a midi controller so you basically have a mixing panel for your different instruments.

```clj
(definst vibe []
  (var-saw (+ 440 (* 20 (sin-osc 30)))))

(inst-volume! vibe 0.8)
(vibe)
```
